src/graccreq/oim/OIMTopology.py

```diff
@@ -62,119 +50,157 @@ def get_file_from_OIM(self):
         return oim_xml
 
     def parse(self):
-        """Parses XML file using ElementTree.parse().  Also sets XML root for
-        further processing"""
+        """Parses XML file using ElementTree.parse().  Also builds dictionary
+        of Resource Name :{resource information} format for future lookups"""
         try:
             self.e = ET.parse(self.xml_file)
             self.root = self.e.getroot()
-            print "Parsed file"
+            print "Parsing file"
         except Exception as e:
             print e
             print "Couldn't parse OIM file"
             self.xml_file = None
+            return
+        
+        for resourcename_elt in self.root.findall('./ResourceGroup/'
+                                                  'Resources/Resource/Name'):
+            resourcename = resourcename_elt.text
+            if resourcename not in self.resourcedict:
+                resource_grouppath = './ResourceGroup/Resources/Resource/' \
+                                     '[Name="{0}"]/../..'.format(resourcename)
+                self.resourcedict[resourcename] = \
+                    self.get_resource_information(resource_grouppath,
```

Comment:
Why do you not pass the resourcename_elt to the `get_resource_information` function?  That would cut down on the root searches, right?